op-service/txmgr/txmgr.go

```diff
@@ -73,6 +69,21 @@ type TxManager interface {
 	// BlockNumber returns the most recent block number from the underlying network.
 	BlockNumber(ctx context.Context) (uint64, error)
 
+	GetMinBaseFee() *big.Int
+	SetMinBaseFee(*big.Int)
+
+	GetPriorityFee() *big.Int
+	SetPriorityFee(*big.Int)
+
+	GetMinBlobFee() *big.Int
+	SetMinBlobFee(*big.Int)
+
+	GetFeeThreshold() *big.Int
+	SetFeeThreshold(*big.Int)
+
+	GetBumpFeeRetryTime() time.Duration
+	SetBumpFeeRetryTime(time.Duration)
+
```

Comment:
I'm wondering if we really need to add all these methods to the `TxManager` interface? I mean, where only the interface is used, are any of these accessed? 
Would be great if we can drop it, so we also don't need to augment the mocks in the challenger and action test.
I see that we pass an interface to `NewTxmgrAPI` because the proposer and challenger work with an interface because of tests. Maybe one solution could be to instead add an method
```suggestion
	API() rpc.API
```
so we let the txmgr return its own API service. This way, in the mocks we can just panic, because afaict it wouldn't be called in tests, only in service setup code. This method would return for the `SimpleTxManager` what `txmgr.NewTxmgrApi` returned before.